sql: remove usages of TODOTestTenantDisabled

yuzefovich opened the pull request:
Addresses: #76378.
Epic: CRDB-18499

yuzefovich committed:
sql: adjust TestValidationWithProtectedTS to work with tenants

This commit adjusts `TestValidationWithProtectedTS` to work with test
tenants too. However, it seems like the test becomes noticeably
slower, and improving the test speed is tracked by #106960.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106883)
<!-- Reviewable:end -->

yuzefovich committed:
sql: enable some tests to run with test tenant

This commit is a grab bag of things:
- remove a couple of TODOTestTenantDisabled in benchmarks to be tracked
by the benchmarking issue
- ensure that all tests in `sql/importer` can run with test tenant
(unless explicitly disabled)
  - enable Export tests
  - adjust `TestProcessorEncountersUncertaintyError`
  - use correct `ExecutorConfig` in some of the tests
  - `TestProtectedTimestampsDuringImportInto` is marked as required
    separate investigation
  - remove updating `storage.mvcc.range_tombstones.enabled` cluster
    setting to `true` since it's already enabled by default on master.

Most of the remaining TODOsin `sql/importer` are currently in `importer`
test package (rather than `importer_test`) which would create an import
cycle if `ccl` is imported, so they will be tackled separately.

Release note: None

yuzefovich committed:
sql: remove some usages of TODOTestTenantDisabled

Three of the tests in `sql` package are explicitly managing the tenants
already.

Release note: None

knz reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with a few clarifying questions

> Reviewed 3 of 3 files at r1, 1 of 1 files at r2, 9 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-:-N_hAhdu30_WyXZ8AYIf:b-yht5t)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stevendanna and @yuzefovich)

___
*[`pkg/sql/backfill_protected_timestamp_test.go` line 101 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_h9lDi4D5-L0IM0N_j:-N_h9lDi4D5-L0IM0N_k:b-jpae0x) ([raw file](https://github.com/cockroachdb/cockroach/blob/08eeb060af6634a568657021869d63128fe4aee9/pkg/sql/backfill_protected_timestamp_test.go#L101)):*
> ```Go
> 	}
> 
> 	// By default, secondary tenants aren't allowed to mock with the zone
> ```

nit: s/mock/muck
___
*[`pkg/sql/conn_executor_test.go` line 1198 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_h8pnJ7f8qwBHUOEy7:-N_h8pnJ7f8qwBHUOEy8:b4no1nd) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa22397287455cec2a782cb80cde08038949239f/pkg/sql/conn_executor_test.go#L1198)):*
> ```Go
> 		},
> 	}
> 	tc := serverutils.StartNewTestCluster(t, 1, testClusterArgs)
> ```

maybe `StartServer` instead of test cluster.
___
*[`pkg/sql/crdb_internal_test.go` line 982 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_h9Bzx08oHQdyu-ZYD:-N_h9Bzy6apF2mZQhb9O:b-sztsn2) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa22397287455cec2a782cb80cde08038949239f/pkg/sql/crdb_internal_test.go#L982)):*
> ```Go
> 
> 	ctx := context.Background()
> 	tc := testcluster.StartTestCluster(t, 1,
> ```

maybe `StartServer` with `TestTenantAlwaysEnabled`, and drop the `StartTenant` call below.
___
*[`pkg/sql/run_control_test.go` line 925 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_h9VnPDBVsXjxGdxGS:-N_h9VnPDBVsXjxGdxGT:b-d5gegp) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa22397287455cec2a782cb80cde08038949239f/pkg/sql/run_control_test.go#L925)):*
> ```Go
> 	}
> 
> 	kvserver, _, _ := serverutils.StartServer(t, params)
> ```

maybe use `TestTenantAlwaysEnabled` and do `tenant := kvserver.TenantOrServer()` below.
___
*[`pkg/sql/run_control_test.go` line 940 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_h9RDt0hhEynukDxlI:-N_h9RDt0hhEynukDxlJ:b-lv735a) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa22397287455cec2a782cb80cde08038949239f/pkg/sql/run_control_test.go#L940)):*
> ```Go
> 
> 	makeTenantConn := func() *sqlutils.SQLRunner {
> 		return sqlutils.MakeSQLRunner(serverutils.OpenDBConn(t, tenant.SQLAddr(), "" /* useDatabase */, false /* insecure */, kvserver.Stopper()))
> ```

that `kvserver.Stopper` here seems wrong, the conn should hang off the tenant stopper instead.
___
*[`pkg/sql/importer/exportparquet_test.go` line 294 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_hAF0K7YPe0WfLixeG:-N_hAF0L5lcmPPj__7ij:bvnseg6) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ec96b38e3fdd0eb51478c9483e54ea9cd3fb0bb/pkg/sql/importer/exportparquet_test.go#L294)):*
> ```Go
> 	sqlDB.Exec(t, fmt.Sprintf("CREATE DATABASE %s", dbName))
> 
> 	// instantiating an internal executor to easily get datums from the table
> ```

nit: capital at start of sentence and period at the end.
___
*[`pkg/sql/importer/import_stmt_test.go` line 138 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_hAT8xB9wGuLeuPqb5:-N_hAT8xB9wGuLeuPqb6:b-j1rrs8) ([raw file](https://github.com/cockroachdb/cockroach/blob/cd0381fc097245106382b4704ab42bf0470c2364/pkg/sql/importer/import_stmt_test.go#L138)):*
> ```Go
> 
> 	sqlDB.Exec(t, `SET CLUSTER SETTING kv.bulk_ingest.batch_size = '10KB'`)
> 	sqlDB.Exec(t, `SET CLUSTER SETTING storage.mvcc.range_tombstones.enabled = true`)
> ```

curious - why did you need to remove this here and below?


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-:-N_jB2EBAsruys7UeqUY:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @knz and @stevendanna)

___
*[`pkg/sql/conn_executor_test.go` line 1198 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_h8pnJ7f8qwBHUOEy7:-N_j4yna3HPXxIZL2jY2:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa22397287455cec2a782cb80cde08038949239f/pkg/sql/conn_executor_test.go#L1198)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

maybe `StartServer` instead of test cluster.
</blockquote></details>

Done.
___
*[`pkg/sql/crdb_internal_test.go` line 982 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_h9Bzx08oHQdyu-ZYD:-N_j5LuX4nPN4c4SR75-:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa22397287455cec2a782cb80cde08038949239f/pkg/sql/crdb_internal_test.go#L982)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

maybe `StartServer` with `TestTenantAlwaysEnabled`, and drop the `StartTenant` call below.
</blockquote></details>

Done.
___
*[`pkg/sql/run_control_test.go` line 925 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_h9VnPDBVsXjxGdxGS:-N_j7P0b9RQ2CYWE-nw9:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa22397287455cec2a782cb80cde08038949239f/pkg/sql/run_control_test.go#L925)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

maybe use `TestTenantAlwaysEnabled` and do `tenant := kvserver.TenantOrServer()` below.
</blockquote></details>

Done.
___
*[`pkg/sql/run_control_test.go` line 940 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_h9RDt0hhEynukDxlI:-N_j7PU23BtpSNk64T_k:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fa22397287455cec2a782cb80cde08038949239f/pkg/sql/run_control_test.go#L940)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

that `kvserver.Stopper` here seems wrong, the conn should hang off the tenant stopper instead.
</blockquote></details>

Done.
___
*[`pkg/sql/importer/import_stmt_test.go` line 138 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/106883#-N_hAT8xB9wGuLeuPqb5:-N_jAZZxCv7eRmkIqIHD:b-edpmjw) ([raw file](https://github.com/cockroachdb/cockroach/blob/cd0381fc097245106382b4704ab42bf0470c2364/pkg/sql/importer/import_stmt_test.go#L138)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

curious - why did you need to remove this here and below?
</blockquote></details>

This setting is already defaulted to `true`, actually not being used on 23.2 binaries, and is tenant read-only so we were getting an error when updating it.


<!-- Sent from Reviewable.io -->
